nistats/tests/test_verify_events_file_uses_tab_separators.py

```diff
@@ -48,16 +51,18 @@ def _run_test_for_invalid_separator(filepath, delimiter_name):
 def test_for_invalid_separator():
     data_for_temp_datafile, delimiters = make_data_for_test_runs()
     for delimiter_name, delimiter_char in delimiters.items():
-        tmp_file_prefix, temp_file_suffix = (
-            'tmp ', ' ' + delimiter_name + '.csv')
-        with NamedTemporaryFile(mode='w', dir=os.getcwd(),
-                                prefix=tmp_file_prefix,
-                                suffix=temp_file_suffix) as temp_csv_obj:
-            _create_test_file(temp_csv=temp_csv_obj,
-                              test_data=data_for_temp_datafile,
-                              delimiter=delimiter_char)
-            _run_test_for_invalid_separator(filepath=temp_csv_obj.name,
-                                            delimiter_name=delimiter_name)
+        tmp_delimited_filename = 'tmp_{}_separated_values_file'.format(delimiter_name)
+        try:
+            with open(tmp_delimited_filename, 'w') as temp_csv_obj:
+                _create_test_file(temp_csv=temp_csv_obj,
+                                  test_data=data_for_temp_datafile,
+                                  delimiter=delimiter_char)
+                _run_test_for_invalid_separator(filepath=temp_csv_obj.name,
+                                                delimiter_name=delimiter_name)
+        except:
```

Comment:
all these `except: raise` are not needed.

Reply:
I replaced the `with NamedTemporaryFile() as ...` with try-except-finally to manually manage the file close and remove. The latter approach was not generating the errors the former was. 
You are suggesting it is the function's implementation that is the source of the problem, which is good news. I will work with you to address the implementation and then restore the former.